cypress/integration/test.js

```diff
@@ -0,0 +1,81 @@
+  //https://min-api.cryptocompare.com/data/pricemulti?fsyms=BTC,ETH,XRP,LTC,BCH,EOS,BNB,ADA,XLM,DOT&tsyms=USD
+  //and store it in a global variable called prices. And display it in the html inside the #prices div.
+  let appPrices;
+  it('fetches data and compares it to stored variable', () => {
+    cy.visit('/')
+    let prices;
+    cy.wait(2000)
+    cy.window().then(win => {
+      prices = win.prices;
+      appPrices = prices;
+    });
+    cy.request('https://min-api.cryptocompare.com/data/pricemulti?fsyms=BTC,ETH,XRP,LTC,BCH,EOS,BNB,ADA,XLM,DOT&tsyms=USD')
+      .then((response) => {
+        expect(response.body).to.deep.equal(prices)
+      })
+  })
+  
+  //Create a global variable called balance which will store the amount of dollars with the user. Its value should be 10000 initally.
+  //Create two global functions handleBuy(e) and handleSel(e). When the user buys crypto by clicking the #buy-button the handleBuy function should 
+  //update the portfolio using the addCryptocurrency function. Similarly when the user sells crypto  by clicking the #sell-button, the handleSell(e)
+  //function should update the portfolio using the removeCryptocurrency function. Both of these function should also update the global variable balance.
+  //At the end, update the UI with the new portfolio data. In the #portfolio div, display the portfolio details in a list, and update the text of
+  //#balance with the new balance, in the form "Balance: $10000"
+
+  it('test buying and selling of crypto', () => {
+    cy.visit('/')
+    cy.wait(2000)
+    cy.get('#currency').select('BNB');
+    cy.get('#units').type(2);
+    cy.get('button[type=submit]').click();
+    cy.wait(100)
+    cy.get('#balance').invoke('text').then((text) => {
+      const balance = text.replace('Balance: $', '');
+      expect(Number(balance)).to.equal(10000 - 2*appPrices['BNB'].USD);
+    });
+
+    cy.get('#currency').select('BNB');
+    cy.get('#units').type(1);
+    cy.get('#sell-button').click();
+    cy.get('#balance').invoke('text').then((text) => {
+      const balance = text.replace('Balance: $', '');
+      expect(Number(balance)).to.equal(10000 - appPrices['BNB'].USD);
+    });
+  })
+
+  //Wheever the portfolio and the balance gets updated, store their data in the localstorage. The portforlio data must to stored in the
+  //key portfolio and the balance of the user must be stored with the key balance. When the page is loaded, initlaize the portfolio and the balance
+  //with the one present in the local storage if present.
+  it('verify local storage', () => {
+    cy.visit('/')
+    cy.wait(2000)
+    cy.get('#currency').select('BNB');
+    cy.get('#units').type(2);
+    cy.get('button[type=submit]').click();
+    cy.window().then((win) => {
+      const balance = win.localStorage.getItem('balance');
+      expect(Number(balance)).to.equal(10000 - 2*appPrices['BNB'].USD);
+    });
+  })
+  
+  
+})
```

Comment:
> please add the testcase for checking if price is lower or higher than purchase price and we are selling, then balance is to be delta +- 10000, this would be by cy.intercept and changing response data

In the "test buying and selling of crypto" testcase, we are already checking if the balance gets updated after selling in the UI. Did you mean we should test the balance variable?

Reply:
I could be wrong but aren't we buying and selling at the same price? we want to check if on day 1 I bought for 10 and and day 2 sell for 20 my balance should be 10k+10